### PR TITLE
Add additional connection and request logging to client

### PR DESCRIPTION
**Pull request by JamesNK**
Additional logging for https://github.com/grpc/grpc-dotnet/issues/1948

Also, implement DisposeAsync on StreamWrapper in as there is a situation where it is called.

### CODE REVIEW

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs [hunk 1]
```diff
@@ -300,6 +309,7 @@ public async ValueTask<Stream> GetStreamAsync(BalancerAddress address, Cancellat
         lock (Lock)
         {
             _activeStreams.Add(new ActiveStream(address, socket, stream));
+            SocketConnectivitySubchannelTransportLog.StreamCreated(_logger, _subchannel.Id, address, _activeStreams.Count);
```
Comment (mgravell) [inline src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:312]: logging inside the lock troubles me; is this intentional? would it perhaps be better to snag the `Count` inside the lock and log outside?

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs [hunk 2]
```diff
@@ -331,7 +341,7 @@ private void OnStreamDisposed(Stream streamWrapper)
                     if (t.Stream == streamWrapper)
                     {
                         _activeStreams.RemoveAt(i);
-                        SocketConnectivitySubchannelTransportLog.DisposingStream(_logger, _subchannel.Id, t.Address);
+                        SocketConnectivitySubchannelTransportLog.DisposingStream(_logger, _subchannel.Id, t.Address, _activeStreams.Count);
```
Comment (mgravell) [inline src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:344]: hmmm, maybe ignore my last comment about lock; I can't see a nice / clean way of avoiding this one, so... if we're in that situation *anyway*, maybe just live with it

**Review by mgravell — APPROVED** [review]
added a question about logging inside the lock, and whether we care about that - I can't get hugely excited either way, as long as considered - looks fine generally

File: src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs (hunk 1 above)
Comment (JamesNK) [reply src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs:312]: I don't think this is a concern. There is logging inside locks throughout gRPC and ASP.NET Core.

**Pull request merged**